frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_property_edit/mobile_field_type_list.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:appflowy/generated/flowy_svgs.g.dart';
+import 'package:appflowy/plugins/database_view/application/field/field_type_option_edit_bloc.dart';
+import 'package:appflowy/plugins/database_view/grid/presentation/widgets/header/field_type_extension.dart';
+
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:appflowy_backend/protobuf/flowy-database2/field_entities.pb.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+typedef SelectFieldCallback = void Function(FieldType);
+
+class MobileFieldTypeList extends StatelessWidget {
+  final FieldTypeOptionEditBloc bloc;
+  final SelectFieldCallback onSelectField;
+  const MobileFieldTypeList({
+    required this.onSelectField,
+    Key? key,
+    required this.bloc,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final cells = FieldType.values.map((fieldType) {
+      return MobileFieldTypeCell(
+        fieldType: fieldType,
+        onSelectField: (fieldType) {
+          onSelectField(fieldType);
+        },
+      );
+    }).toList();
+
+    return BlocProvider.value(
+      value: bloc,
+      child: ListView.builder(
+        shrinkWrap: true,
+        itemCount: cells.length,
+        itemBuilder: (_, index) {
+          return cells[index];
+        },
+      ),
+    );
+  }
+}
+
+class MobileFieldTypeCell extends StatelessWidget {
+  final FieldType fieldType;
+  final SelectFieldCallback onSelectField;
+  const MobileFieldTypeCell({
+    required this.fieldType,
+    required this.onSelectField,
+    Key? key,
+  }) : super(key: key);
```

Comment:
Personal preference + super.key constructor
```suggestion
  const MobileFieldTypeCell({
    super.key,
    required this.fieldType,
    required this.onSelectField,
  });

  final FieldType fieldType;
  final SelectFieldCallback onSelectField;
```